old_api/__init__.py

```diff
@@ -79,15 +79,19 @@ def upload_file(release_id, relpath, abspath, username):
     )
 
     if response.status_code != 201:
-        logger.error(
-            "%s Error uploading file - %s - %s - %s",
-            response.status_code,
-            relpath,
-            release_id,
-            response.content.decode(),
-        )
-
-    response.raise_for_status()
+        response_content = response.content.decode()
+        if "has already been uploaded" in response_content:
```

Comment:
This should be able to be a more explicit check, as the API response is json?

e.g. something like `response.json()["detail"] == ...`

Reply:
I've made this more explicit, but it can't (without a job-server change) check the exact message, because it contains the backend name, and airlock doesn't itself know what it's called on job-server.